hecuba_py/src/hdict.py

```diff
@@ -529,6 +536,21 @@ def __getitem__(self, key):
             else:
                 return final_results[0]
 
+    def __make_val_persistent(self, val, col=0):
+        if isinstance(val, StorageDict):
+            for k, element in val.iteritems():
```

Comment:
Why? Why don't you just check if the object is already persistent (then do nothing) and otherwise you call val.make_persistent(....) ?